syntax = "proto3";

package map_renderer_serialize;

message Rgb {
    uint32 red = 1;
    uint32 green = 2;
    uint32 blue = 3;
}

message Rgba {
    uint32 red = 1;
    uint32 green = 2;
    uint32 blue = 3;
    double opacity = 4;
}

message Color {
    oneof data {
        bool is_none = 1;
        string name = 2;
        Rgb rgb = 3;
        Rgba rgba = 4;
    }
}

message MapRenderer {
    double width = 1;
    double height = 2;
    double padding = 3;
    double line_width = 4;
    double stop_radius = 5;
    uint32 bus_label_font_size = 6;
    double bus_label_offset_x = 7;
    double bus_label_offset_y = 8;
    uint32 stop_label_font_size = 9;
    double stop_label_offset_x = 10;
    double stop_label_offset_y = 11;
    Color underlayer_color = 12;
    double underlayer_width = 13;
    repeated Color color_palette = 14;
}
